ludwig/models/llm.py

```diff
@@ -505,7 +509,22 @@ def update_metrics_finetune_llm(self, targets, predictions):
                 # to match the prediction length and depends on how much of the target tensor was included in the
                 # forward pass.
                 _targets = self._update_target_tensor_for_finetuning(_targets, _predictions, of_name)
-                of_obj.update_metrics(_targets[of_name], _predictions[of_name])
+                if isinstance(of_obj, TextOutputFeature):
+                    # NOTE: Alignment is not technically necessary here, but we re-use the method used in
+                    # LLM.update_metrics() for decoding consistency.
+                    (
+                        _targets,
+                        _decoded_targets,
+                        _predictions,
+                        _decoded_predictions,
+                    ) = realign_target_and_prediction_tensors_for_inference(
+                        _targets, _predictions, of_name, self.tokenizer
+                    )
+                    of_obj.update_metrics(
```

Comment:
like above, this happens in at least two palces then

Reply:
Acknowledged.